tests/end_to_end/end_to_end.cpp

```diff
@@ -673,3 +680,106 @@ TEST_CASE("pinned_map_enum", "[end_to_end]")
 
     ebpf_test_pinned_map_enum();
 }
+
+TEST_CASE("implicit_explicit_detach", "[end_to_end]")
+{
+    _test_helper_end_to_end test_helper;
+
+    ebpf_handle_t program_handle;
+    ebpf_handle_t map_handle;
+    uint32_t count_of_map_handle = 1;
+    uint32_t result = 0;
+    const char* error_message = NULL;
```

Comment:
```suggestion
    const char* error_message = nullptr;
```
use nullptr in cpp files